packages/pigment-css-react/README.md

```diff
@@ -204,11 +208,15 @@ Each variant is an object with `props` and `style` keys. The styles are applied
 
 **Example 1** — A button component with `small` and `large` sizes:
 
-```jsx
-const Button = styled('button')({
+```tsx
+interface ButtonProps {
+  size?: 'small' | 'large';
+}
+
+const Button = styled('button')<ButtonProps>({
```

Comment:
So, I started out with the Vite example app and just copying and pasting this code snippet doesn't work because of TypeScript. Given both example apps (Vite & Next.js) are using it by default, I wonder if we shouldn't do something like this? Y'all tell me if this is the best way to do this, as I know there are others. But I guess having these snippets tailored to work seamlessly with the example apps should be the way to go.

This point was what motivated me to open the PR. Depending on what we discuss here, we can change other snippets throughout the README!

Reply:
Okay, I just realized (sorry 😅) we have the "Typing props" header a few lines below this code snippet. This sort of solves my point there, but I do still wonder if we shouldn't tailor the docs to TypeScript by default and warn the opposite: instead of saying "If you use TS, add this", say "if you're not using TS, remove this"?

Reply:
Definitely—I can also do the opposite (add a comment about heading over to the TS adaptation) if we're not in the mood to debate making them use TypeScript by default right now. I personally feel like it'd be good to do it, though... the example apps use it, and our own codebase also largely uses it, so I think the consistency would be great there.

Reply:
We do have a Typescript section in this README. So we can target ts there. Anyways, there are only 2 points to note for TS (as of now). Rest can just target JS even for TS code.
I am OK with either. One data point to figure out would how many people report issues in our repo specific to writing code for JS and not TS.